### PR TITLE
feat!: implement artifact manifest handler in the handler package

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Part 4 of #21.

Implemented necessary changes in the handler package.

Signed-off-by: wangxiaoxuan273 <wangxiaoxuan119@gmail.com>

### CODE REVIEW

**Comment by wangxiaoxuan273** [comment]
To test on terminal:
* push a minimal artifact manifest: `curl -X PUT -v localhost:5000/v2/mantest/manifests/bc8a5e80bcb03fe9b5d74db83b1c47e6f7e6b7ca -H "Content-Type:application/vnd.oci.artifact.manifest.v1+json" -d '{"mediaType": "application/vnd.oci.artifact.manifest.v1+json"}'`

File: registry/handlers/manifests.go [hunk 1]
```diff
@@ -80,9 +83,9 @@ type manifestHandler struct {
 	Digest digest.Digest
 }
 
-// GetManifest fetches the image manifest from the storage backend, if it exists.
+// GetManifest fetches the manifest from the storage backend, if it exists.
 func (imh *manifestHandler) GetManifest(w http.ResponseWriter, r *http.Request) {
-	dcontext.GetLogger(imh).Debug("GetImageManifest")
+	dcontext.GetLogger(imh).Debug("GetManifest")
```
Comment (shizhMSFT) [inline registry/handlers/manifests.go:88]: Nice catch! LoL

File: registry/handlers/manifests.go [hunk 2]
```diff
@@ -322,7 +335,13 @@ func (imh *manifestHandler) PutManifest(w http.ResponseWriter, r *http.Request)
...
 	}
 
-	isAnOCIManifest := mediaType == v1.MediaTypeImageManifest || mediaType == v1.MediaTypeImageIndex
+	var isAnOCIManifest bool
+	switch mediaType {
+	case v1.MediaTypeImageManifest, v1.MediaTypeArtifactManifest, v1.MediaTypeImageIndex:
+		isAnOCIManifest = true
+	default:
+		isAnOCIManifest = false
+	}
```
Comment (shizhMSFT) [inline registry/handlers/manifests.go:344]: nit: simplify code 😄  
```suggestion
	isAnOCIManifest := false
	switch mediaType {
	case v1.MediaTypeImageManifest, v1.MediaTypeArtifactManifest, v1.MediaTypeImageIndex:
		isAnOCIManifest = true
	}
```

File: registry/handlers/manifests.go [hunk 3]
```diff
@@ -439,6 +458,13 @@ func (imh *manifestHandler) applyResourcePolicy(manifest distribution.Manifest)
 		default:
 			return errcode.ErrorCodeDenied.WithMessage("unknown manifest class for " + m.Config.MediaType)
 		}
+	case *ociartifact.DeserializedManifest:
+		switch m.MediaType {
+		case v1.MediaTypeArtifactManifest:
+			class = artifactClass
+		default:
+			return errcode.ErrorCodeDenied.WithMessage("unknown manifest class for " + m.MediaType)
+		}
```
Comment (shizhMSFT) [inline registry/handlers/manifests.go:467]: The `m.MediaType` is always `v1.MediaTypeArtifactManifest`.
```suggestion
		class = artifactClass
```

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: registry/handlers/manifests.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply registry/handlers/manifests.go:344]: changed accordingly.

File: registry/handlers/manifests.go (hunk 3 above)
Comment (wangxiaoxuan273) [reply registry/handlers/manifests.go:467]: changed accordingly.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**